contracts/amm.cairo

```diff
@@ -210,11 +221,21 @@ func close_position{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_chec
 
     let (opposite_side) = get_opposite_side(side);
 
-    let (option_size_in_pool_currency) = convert_amount_to_option_currency_from_base(
-        option_size,
+    let option_size_uint256 = intToUint256(option_size);
+    let strike_price_uint256 = toUint256_balance(strike_price, quote_token_address);
+    let (option_size_in_pool_currency) = convert_amount_to_option_currency_from_base_uint256(
+        option_size_uint256,
         option_type,
-        strike_price
+        strike_price_uint256,
+        base_token_address
     );
+    assert option_size_in_pool_currency.high = 0;
+
+    let option_size_m64x61_ = Math64x61.fromFelt(option_size);
+    let (base_decimals: felt) = IERC20.decimals(contract_address=base_token_address);
+    let (base_div) = pow10(base_decimals);
+    let base_div_m64x61 = Math64x61.fromFelt(base_div);
+    let option_size_m64x61 = Math64x61.div(option_size_m64x61_, base_div_m64x61);
```

Comment:
are there some assurances that this won't overflow, ie if the option size is in tens of thousands? Since the div will multiply the x by fract again, if the option would be for DOGE(8 decimals afaik), then the size of 20_000 would be sth like -> 20_000 * 10**8 * FRACT_PART(first 'fromFelt') * FRACT_PART(inside 'div') which

Reply:
ETH has 18 decimals and it isn't a problem. m64x61 fails on overflow

Reply:
okey